argon2/src/memory.rs

```diff
@@ -0,0 +1,162 @@
+//! Views into Argon2 memory that can be processed in parallel.
+//!
+//! This module implements, with a combination of compile-time borrowing and runtime checking, the
+//! cooperative contract described in section 3.4 (Indexing) of RFC 9106:
+//!
+//! > To enable parallel block computation, we further partition the memory matrix into SL = 4
+//! > vertical slices. The intersection of a slice and a lane is called a segment, which has a
+//! > length of q/SL. Segments of the same slice can be computed in parallel and do not reference
+//! > blocks from each other. All other blocks can be referenced.
+
+#![warn(
+    clippy::undocumented_unsafe_blocks,
+    clippy::missing_safety_doc,
```

Comment:
Hmm, I guess we don't have these configured at the toplevel? It would probably be good to get them there eventually

Reply:
Sure, sounds good